### PR TITLE
Fix waterbox off behaviour (#764)

### PR DESCRIPTION
**Pull request by afharo**
Resolves https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/issues/764#issuecomment-1712083333

### CODE REVIEW

File: src/services/water_box_service.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class WaterBoxService extends PluginServiceClass {
       .onSet(async (newState) => {
         // Set to 0% (Off) when receiving an OFF request, do nothing otherwise.
         if (!newState) {
-          await this.mainService.setCleaning(false);
+          await this.setWaterSpeed(0);
```
Comment (afharo) [inline src/services/water_box_service.ts:37]: this is the actual fix

**Comment by afharo** [comment]
It should be fixed in https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/releases/tag/v0.29.1. Please, let me know if it isn't.

**Comment by pelletip** [comment]
It does not work, at least for me, with my Q Revo.  The mop spins and the floor gets mopped, even with the waterbox set to off.  Changing the level on the waterbox does not seem to change the amount of water getting fed to the mops either, but I may be wrong given my limited observations testing that portion of the fonctionalities. 

Screenshot 2023-09-08 at 7 05 37 PM

**Comment by afharo** [comment]
@pelletip, do you get any errors in the logs when setting the waterbox to off?

**Comment by pelletip** [comment]
No errors listed in the logs.
Also, it does not look like there is an option to send the robot on a "mopping only" routine.  If I press the waterbox button, it does not start... I need to press the Robot Vacuum button for it to get started, once it starts, it's always the combo (mop and vacuum) running

**Comment by pelletip** [comment]
wait, I clicked the plugin option to see more logs:  some waterbox related logs: 
[08/09/2023, 19:22:12] [Robot Vaccum] [Model=roborock.vacuum.a75] getWaterSpeed | WaterBoxMode is undefined over miIO. Converting to HomeKit
[08/09/2023, 19:22:12] [Robot Vaccum] [Model=roborock.vacuum.a75] getWaterSpeed | WaterBoxMode is undefined over miIO. Converting to HomeKit

**Comment by afharo** [comment]
yeah... I think that model didn't have any listed WaterBox-compatible in modes in the plugin. I hope [v0.30.0](https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/releases/tag/v0.30.0) fixes it for you

**Comment by pelletip** [comment]
it definitely is better, I can now do vacuum only by setting the 'water box' button 'off'.  Thank you so much for fixing!   I guess there is no option to do a mop only routine?  the only way to initiate a cleaning is to press the vacuum button, if I slide the vacuum to zero, it returns to the dock.

**Comment by afharo** [comment]
I can look into that. I created https://github.com/homebridge-xiaomi-roborock-vacuum/homebridge-xiaomi-roborock-vacuum/issues/791 to track this additional request :)

**Comment by pelletip** [comment]
maybe a moot point anyways... it's probably a good idea to at least have minimal vacuum ahead of mopping... I assume it can be done technically, because the Mi app allows for this type of routine to be executed, but this may be a bit of a rare use case, comes to think of it.

**Pull request merged**